dev-scripts/deploy_custom_rh-che.sh

```diff
@@ -0,0 +1,304 @@
+                     .\"data\".\"che.jdbc.url\" = \"$RH_CHE_JDBC_URL\" | 
+                     .\"metadata\".\"name\" = \"che\" " ./rh-che.config.yaml)
+
+#echo "$CHE_CONFIG_YAML"
+if ! (echo "$CHE_CONFIG_YAML" | oc apply -f - 2>1 > /dev/null); then
+  echo -e "\e[91;1mFailed to apply configmap.\e[0m"
+  exit 1
+fi
+echo -e "\e[92;1mChe config deployed on \e[34m$RH_CHE_PROJECT_ID\e[0m"
+
+# PROCESS CHE APP CONFIG
+sed -i 's/rhche/che/g' ./rh-che.app.yaml
+#oc process -f ./rh-che.app.yaml | cat
+CHE_APP_CONFIG_YAML=$(yq "(.objects[] | select(.kind == \"DeploymentConfig\").spec.template.spec.containers[0].env[] | 
+                         select(.name == \"CHE_OPENSHIFT_SERVICE__ACCOUNT_SECRET\").valueFrom) |= {\"configMapKeyRef\":{\"key\":\"service.account.secret\",\"name\":\"che\"}} | 
+                         (.objects[] | select(.kind == \"DeploymentConfig\").spec.template.spec.containers[0].env[] | 
+                         select(.name == \"CHE_OPENSHIFT_SERVICE__ACCOUNT_ID\").valueFrom) |= {\"configMapKeyRef\":{\"key\":\"service.account.id\",\"name\":\"che\"}} | 
+                         (.objects[] | select(.kind == \"DeploymentConfig\").spec.template.spec.containers[0].env[] | 
+                         select(.name == \"CHE_JDBC_PASSWORD\").valueFrom) |= {\"configMapKeyRef\":{\"key\":\"che.jdbc.password\",\"name\":\"che\"}} | 
+                         (.objects[] | select(.kind == \"DeploymentConfig\").spec.template.spec.containers[0].env[] | 
+                         select(.name == \"CHE_JDBC_URL\").valueFrom) |= {\"configMapKeyRef\":{\"key\":\"che.jdbc.url\",\"name\":\"che\"}} | 
+                         (.objects[] | select(.kind == \"DeploymentConfig\").spec.template.spec.containers[0].env[] | 
+                         select(.name == \"CHE_JDBC_USERNAME\").valueFrom) |= {\"configMapKeyRef\":{\"key\":\"che.jdbc.username\",\"name\":\"che\"}} | 
+                         (.parameters[] | select(.name == \"IMAGE\").value) |= \"$RH_CHE_DOCKER_IMAGE_REGISTRY\" | 
+                         (.parameters[] | select(.name == \"IMAGE_TAG\").value) |= \"$RH_CHE_PROJECT_ID\" " ./rh-che.app.yaml)
+
+#echo "$CHE_APP_CONFIG_YAML" | oc process -f - | cat
+if ! (echo "$CHE_APP_CONFIG_YAML" | oc process -f - | oc apply -f - 2>1 > /dev/null); then
+  echo -e "\e[91;1mFailed to process che config\e[0m"
+  exit 1
+fi
+
+CHE_STARTUP_TIMEOUT=120
+while [[ "${RH_CHE_STATUS_PROGRESS}" != "\"True\"" || "${RH_CHE_STATUS_AVAILABLE}" != "\"True\"" ]] && [ ${CHE_STARTUP_TIMEOUT} -gt 0 ]; do
+  sleep 1
+  checkCheStatus
+  echo -e "\e[0;38;5;60mChe-server status: Available:\e[0;1m$RH_CHE_STATUS_AVAILABLE \e[0;38;5;60mProgressing:\e[0;1m$RH_CHE_STATUS_PROGRESS \e[0;38;5;60mTimeout:\e[0;1m$CHE_STARTUP_TIMEOUT\e[0m"
+  CHE_STARTUP_TIMEOUT=$((CHE_STARTUP_TIMEOUT-1))
+done
+if [ $CHE_STARTUP_TIMEOUT == 0 ]; then
+  echo -e "\e[91;1mFailed to start che-server: timed out\e[0m"
+  exit 1
+fi
+
+oc annotate --overwrite=true route/che kubernetes.io/tls-acme=true
```

Comment:
What do you need this annotation for?

Reply:
this one is required for enabling valid acme certificate for che server, otherwise ws-agent -> wsmaster communication would fail due to untrusted cert